cloudpickle/cloudpickle.py

```diff
@@ -580,6 +576,7 @@ def save_dynamic_class(self, obj):
 
         # If type overrides __dict__ as a property, include it in the type
         # kwargs. In Python 2, we can't set this attribute after construction.
+        # XXX: can this ever happen in Python 3? If so add a test.
         __dict__ = clsdict.pop('__dict__', None)
         if isinstance(__dict__, property):
             type_kwargs['__dict__'] = __dict__
```

Comment:
Not so sure about this: `type_kwargs['__dict__'] = __dict__` is no longer covered by our tests.